backend/pkg/sqlretry/dbtx_retry.go

```diff
@@ -107,6 +107,35 @@ func (r *RetryDBTX) BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx,
 	return retry(ctx, operation, r.config.getRetryOpts)
 }
 
+func (r *RetryDBTX) RetryTx(ctx context.Context, opts *sql.TxOptions, fn func(*sql.Tx) error) error {
```

Comment:
This is more useful than the BeginTx above as it handles the transaction lifecycle for you.